### DISCUSSION TITLE
Do I still need argo-rollouts if I use AWS App mesh

### DISCUSSION THREAD

**Discussion by DavidHe1127** [argoproj/argo-rollouts#2420] [Q&A]
I see both solutions have crossovers in some things like blue/green, canary deployment strategies. My silly question is do what extra benefits argo-rollout provides than AWS App mesh?

**Comment by kostis-codefresh** [comment]
Argo Rollouts is not a networking solution. It is a deployment solution.

It actually integrates with several traffic providers for the actual routing.

I just saw that app mesh support was recently added https://github.com/argoproj/argo-rollouts/pull/1606/files

Example is here https://github.com/argoproj/argo-rollouts/tree/master/examples/appmesh

**Reply by DavidHe1127** [reply]
Thanks for the quick reply. I can now see the diff. If I want to have a support of blue/green and canary deployment solutions, I could use AWS App Mesh without argo rollout. Is it correct?

**Reply by kostis-codefresh** [reply]
I am not an expert on aws mesh, but if it is like other meshes it is only a networking solution. You also need Argo rollouts for progressive delivery
